### PR TITLE
Fall back to `origin` for fetch and push

### LINKED ISSUE #53: git_push() and git_fetch() should have a default remote
I think `git_push()` should have some default logic re: `remote`. What should it be,  though?

Docs for `git push`: <https://git-scm.com/docs/git-push>

> When the command line does not specify where to push with the `<repository>` argument, `branch.*.remote configuration` for the current branch is consulted to determine where to push. If the configuration is missing, it defaults to `origin`.

Seems like `gert::git_push()` should match that as closely as possible. The existing check against `git_info()$remote` appears to implement the check against `branch.*.remote configuration` for the current branch.

https://github.com/r-lib/gert/blob/4fe319dc8260072c46c048f5f56b04aef3791763/R/fetch.R#L50-L53

But this guard feels too aggressive:

https://github.com/r-lib/gert/blob/4fe319dc8260072c46c048f5f56b04aef3791763/R/fetch.R#L55-L56

I think this should fall back to `origin`.

I'm about to post a reprex showing how `git_push()` insists on me specifying a remote, even in a case where `git push` in the shell succeeds (falling back to `origin`).

### PR DESCRIPTION
**Pull request by jennybc**
Closes #53

### CODE REVIEW

File: R/fetch.R [hunk 1]
```diff
@@ -19,11 +21,14 @@ git_fetch <- function(remote = NULL, refspec = NULL, password = askpass,
                       ssh_key = NULL, verbose = interactive(), repo = '.'){
   repo <- git_open(repo)
   info <- git_info(repo)
+
+  remote <- remote %||% info$remote
```
Comment (jeroen) [inline R/fetch.R:25]: I don't think we have `%||%`

Comment (jeroen) [reply R/fetch.R:25]: Edit: oh wait I see it now at the bottom :)

**Comment by jeroen** [comment]
But in the command line, if you create a new repo, and then you do `git push` I see:

```
fatal: The current branch master has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin master
```

Even if a remote called `origin` exists.

**Comment by jennybc** [comment]
> But in the command line, if you create a new repo, and then you do `git push` I see:
> 
> ```
> fatal: The current branch master has no upstream branch.
> To push the current branch and set the remote as upstream, use
> 
>     git push --set-upstream origin master
> ```
> 
> Even if a remote called `origin` exists.

And this is in a repo with commits? I.e. `master` isn't unborn or anything weird like that? Because as shown above, I can push with `git push` in this case.

I should also mention that I have `push.default` set to `current` in my global git config file. But I didn't really think that had an influence over this specific situation. *Update: OK there is a situation in which `push.default` matters = when neither the remote repo nor the ref spec is given, so an argument-less `git push`*:

> When the command line does not specify what to push with `<refspec>...` arguments or `--all`, `--mirror`, `--tags` options, the command finds the default `<refspec>` by consulting `remote.*.push` configuration, and if it is not found, honors `push.default` configuration to decide what to push (See git-config[1] for the meaning of `push.default`).

File: R/fetch.R (hunk 1 above)
Comment (jennybc) [reply R/fetch.R:25]: Yeah, I can remove that and spell it out if you wish. But I use this operator all the time (I usually take it from rlang).

**Comment by jeroen** [comment]
Ah ok that explains. I'm fine with making gert behave like `push.default: current`. Just for reference, in the default git configuration I see this:


```sh
# Init new repo
mkdir test
cd test
git init

# Create a commit
echo "bla" > foo.txt
git add foo.txt
git commit -m 'This is a test commit'

# add an origin remote
git remote add origin https://github.com/jeroen/test
git push
```

That gives me:


```
fatal: The current branch master has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin master
```

**Comment by jennybc** [comment]
For completeness and posterity, here are the docs for `push.default`:

https://git-scm.com/docs/git-config#Documentation/git-config.txt-pushdefault

I have configured this (intentionally) to `current`, inspired by my last deep dive into all of this behaviour when I was working usethis PR stuff.

> `current` - push the current branch to update a branch with the same name on the receiving end. Works in both central and non-central workflows.

But it is true that the default is ?I think? `simple` and that must give the behaviour you're seeing @jeroen.

It does feel pretty safe / unsurprising for `git_push()` and `git_fetch()` to fall back to `origin`, though. I don't think this proposal is very radical.

File: R/fetch.R (hunk 1 above)
Comment (jeroen) [reply R/fetch.R:25]: Yes I prefer to use base-r syntax like in the original code. I always use `NULL` as default icw/ this pattern:

```r
  if(!length(remote))
   remote <- info$remote
```

**Pull request closed without merging**